test-fork/incentives-revision-2-upgrade.spec.ts

```diff
@@ -0,0 +1,194 @@
+      AAVE_DAI_TOKEN
+    )) as IERC20;
+    dai = (await ethers.getContractAt(
+      '@aave/protocol-v2/contracts/dependencies/openzeppelin/contracts/IERC20.sol:IERC20',
+      DAI_TOKEN
+    )) as IERC20;
+    stkAave = (await ethers.getContractAt(
+      '@aave/protocol-v2/contracts/dependencies/openzeppelin/contracts/IERC20.sol:IERC20',
+      STAKED_AAVE
+    )) as IERC20;
+
+    // Define system contracts
+    incentives = (await ethers.getContractAt(
+      'StakedTokenIncentivesController',
+      INCENTIVES_PROXY
+    )) as IAaveIncentivesController;
+    incentivesProxy = (await ethers.getContractAt(
+      'InitializableAdminUpgradeabilityProxy',
+      INCENTIVES_PROXY
+    )) as InitializableAdminUpgradeabilityProxy;
+    const poolAddressesProvider = (await ethers.getContractAt(
+      '@aave/protocol-v2/contracts/interfaces/ILendingPoolAddressesProvider.sol:ILendingPoolAddressesProvider',
+      AAVE_LENDING_POOL_ADDRESSES_PROVIDER
+    )) as ILendingPoolAddressesProvider;
+    pool = (await ethers.getContractAt(
+      'ILendingPool',
+      await poolAddressesProvider.getLendingPool()
+    )) as ILendingPool;
+
+    // Spoof signers for the necessary accounts
+    await impersonateAccountsHardhat([INCENTIVES_PROXY_ADMIN, DAI_WHALE]);
+    daiWhale = ethers.provider.getSigner(DAI_WHALE);
+
+    // Seed the impersonated accounts with some ETH for gas
+    for (const ethBeneficiary of [INCENTIVES_PROXY_ADMIN, DAI_WHALE]) {
+      const selfDestructContract = await deploySelfDestruct();
+      await selfDestructContract.destroyAndTransfer(ethBeneficiary, {
+        value: ethers.utils.parseEther('10'),
+      });
+    }
+
+    // Deploy next incentives implementation
+    const nextIncentivesImplementation = await deployAaveIncentivesController([
+      STAKED_AAVE,
+      AAVE_SHORT_EXECUTOR,
+    ]);
+    nextIncentivesImplementationAddress = nextIncentivesImplementation.address;
+
+    rewardsAssets = [aDai.address];
+  });
+
+  it('pre-upgrade: accrues rewards', async () => {
+    // Seed claimer with DAI
+    const daiDepositAmount = ethers.utils.parseEther('10000');
+    await dai.connect(daiWhale).transfer(claimer.address, daiDepositAmount);
+
+    // Deposit DAI for aDAI
+    await dai.connect(claimer).approve(pool.address, daiDepositAmount);
+    await pool.connect(claimer).deposit(dai.address, daiDepositAmount, claimer.address, 0);
+    expect(await aDai.balanceOf(claimer.address)).to.be.gt(0);
+
+    // Claim all rewards to zero out previously-accrued rewards
+    await incentives
+      .connect(claimer)
+      .claimRewards(rewardsAssets, constants.MaxUint256, claimer.address);
+
+    // Wait for some time to accrue rewards
+    await increaseTime(rewardsAccrualDuration);
+    preUpgradeRewardsBalance = await incentives.getRewardsBalance(rewardsAssets, claimer.address);
+    expect(preUpgradeRewardsBalance).to.be.gt(0);
+  });
+
+  it('upgrades', async () => {
```

Comment:
In order to test the whole process, would be nice to run the gov proposal process here (no impersonating).
Then, I think would be nice to move the process of the other test file here.